py/parse.c

```diff
@@ -427,6 +417,25 @@ STATIC mp_obj_t mp_parse_node_convert_to_obj(mp_parse_node_t pn) {
 }
 #endif
 
+STATIC bool parse_node_is_const_bool(mp_parse_node_t pn, bool value) {
+    // Returns true if 'pn' is a constant whose boolean value is equivalent to 'value'
+    #if MICROPY_COMP_CONST_TUPLE || MICROPY_COMP_CONST
+    return mp_parse_node_is_const(pn) && (bool)mp_obj_is_true(mp_parse_node_convert_to_obj(pn)) == value;
```

Comment:
I've not seen `(bool)` casting used like this before... is it the same as `!!(..)`?

Reply:
yes, the same - it's to avoid implicit promotion of the bool to an int (on the RHS). if you prefer `!!` then can use this.

Reply:
Switched to `!!` in line with out usage in MP